src/remote/configuration/__init__.py

```diff
@@ -117,3 +117,9 @@ def is_workspace_root(self, path: Path) -> bool:
     @abstractmethod
     def generate_remote_directory(self, config: WorkspaceConfig) -> Path:
         """Renerate a default remote directory path for the workspace with provided configuration"""
+
+
+@dataclass
+class PortForwardingConfig:
+    remote_port: int
+    local_port: int
```

Comment:
I feel that this class is unnecessary here. It represents the CLI input option, not a workspace configuration. Also, it might be replaced by a tuple for simplicity in ssh function input